src/library_loader.h

```diff
@@ -0,0 +1,115 @@
+/*
```

Comment:
I think the library loader should at least be its own commit, if not its own PR.

Reply:
Sounds like a good idea!